cloud-consumption/src/main/java/com/sequenceiq/consumption/job/storage/StorageConsumptionJobService.java

```diff
@@ -0,0 +1,109 @@
+
+    private static final Random RANDOM = new SecureRandom();
+
+    @Inject
+    private  ApplicationContext applicationContext;
+
+    @Inject
+    private Scheduler scheduler;
+
+    @Inject
+    private StorageConsumptionConfig storageConsumptionConfig;
+
+    public void schedule(StorageConsumptionJobAdapter resource) {
+        if (storageConsumptionConfig.isStorageConsumptionEnabled()) {
+            JobDetail jobDetail = buildJobDetail(resource);
+            Trigger trigger = buildJobTrigger(jobDetail);
+            try {
+                JobKey jobKey = JobKey.jobKey(resource.getJobResource().getLocalId(), JOB_GROUP);
+                if (scheduler.getJobDetail(jobKey) != null) {
+                    LOGGER.info("Unscheduling storage consumption job key: '{}' and group: '{}'", jobKey.getName(), jobKey.getGroup());
+                    unschedule();
+                }
+                LOGGER.info("Scheduling storage consumption job for key: '{}' and group: '{}'", jobKey.getName(), jobKey.getGroup());
+                scheduler.scheduleJob(jobDetail, trigger);
+            } catch (SchedulerException e) {
+                LOGGER.error(String.format("Error during scheduling quartz job: %s", jobDetail), e);
+            }
+        }
+
+    }
+
+    public void schedule(Long id) {
+        StorageConsumptionJobAdapter resourceAdapter = new StorageConsumptionJobAdapter(id, applicationContext);
+        schedule(resourceAdapter);
+    }
+
+    private JobDetail buildJobDetail(StorageConsumptionJobAdapter resource) {
+        JobDataMap jobDataMap = resource.toJobDataMap();
+        return JobBuilder.newJob(StorageConsumptionJob.class)
+                .withIdentity(resource.getJobResource().getLocalId(), JOB_GROUP)
+                .withDescription("Getting storage usage")
+                .usingJobData(jobDataMap)
+                .storeDurably()
+                .build();
+
+    }
+
+    private Trigger buildJobTrigger(JobDetail jobDetail) {
+        return TriggerBuilder.newTrigger()
+                .forJob(jobDetail)
+                .withIdentity(jobDetail.getKey().getName(), TRIGGER_GROUP)
+                .withDescription("Trigger for getting storage usage")
+                .startAt(delayedFirstStart())
+                .withSchedule(SimpleScheduleBuilder.simpleSchedule()
+                        .withIntervalInMinutes(storageConsumptionConfig.getIntervalInMinutes())
+                        .repeatForever()
+                        .withMisfireHandlingInstructionIgnoreMisfires())
+                .build();
+
+    }
+
+    public void unschedule() {
+        JobKey jobKey = JobKey.jobKey(JOB_NAME, JOB_GROUP);
+        try {
+            scheduler.deleteJob(jobKey);
+        } catch (SchedulerException e) {
+            LOGGER.error(String.format("Error during unscheduling quartz job: %s", jobKey), e);
+        }
+
+    }
+
+    private Date delayedFirstStart() {
```

Comment:
Delaying by minutes might not have a good distribution (i.e. in the case of normal distribution would mean on average if we have more than 30 buckets to watch then more than one would be started at the same second).